src/parser.ts

```diff
@@ -37,6 +38,10 @@ export function parse(
   usedNames = new Set<string>()
 ): AST {
   if (isPrimitive(schema)) {
+    if (isBoolean(schema)) {
+      return parseBooleanSchema(schema, keyName, options)
+    }
+
     return parseLiteral(schema, keyName)
```

Comment:
tbh I don't know what to do here. In theory we should omit anything at this point instead of parsing the literal, but I don't want to introduce too many changes in this PR

Reply:
Could be a little cleaner to move the `parseBooleanSchema` into `parseLiteral`, since `true`/`false` are literals.

Reply:
I think this would be a mistake. Parsing `true` and `false` as schemas is not the same as parsing them as values. Depending on the context, `true` and `false` are schemas or values.